### ISSUE TITLE
Kernel: addition overflow on long int in Thread::BlockTimeout constructor

### ISSUE THREAD

**Issue by Lubrsi** [SerenityOS/serenity#5315]
Found with fuzz-syscalls.

Trace:
```
fuzz-syscalls(58): #6620: Calling poll(0x115a8000, 0xc0000000, 0xbffff000) with 0x115a8000 containing [0xffffffff, 0x00000000, 0xa405a98b, 0xc0000000, 0xbffff000, 0xbffff000, 0xc0000000, 0x00000001, 0x00000000]
[#0 H(58:58)]: KUBSAN: addition overflow, 'long int' (32-bit)
[#0 H(58:58)]: KUBSAN: at .././AK/Time.h, line 117, column: 32
[#0 H(58:58)]: 0xc022302a  print_location +0xfe
[#0 H(58:58)]: 0xc01e0744  Kernel::Thread::BlockTimeout::BlockTimeout(bool, timespec const*, timespec const*, int) +0x1a8
[#0 H(58:58)]: 0xc01f77eb  Kernel::Process::sys$poll(AK::Userspace<Kernel::Syscall::SC_poll_params const*>) +0x1b9
[#0 H(58:58)]: 0xc01ccda8  syscall_handler +0xa87
[#0 H(58:58)]: 0xc01cc2e9  syscall_asm_entry +0x31
```

**Comment by BenWiederhake** [comment]
This seems already fixed in 62f0f73bf0e15e9830486a51c2114ae8399dbade, can you confirm?

**Comment by BenWiederhake** [comment]
Fixed by #5323. You can verify this by running `syscall poll [ 0xffffffff 0 "Hello, world!" 0xc0000000 ]`.

**Comment by IdanHo** [comment]
Closed as fixed (see above)
